### DISCUSSION TITLE
[OFFICIAL] Talk about Development

### DISCUSSION THREAD

**Discussion by BaseMax** [On-System/On#1] [Show and tell]
Hi,

The **On System** is not yet ready and complete and we need the help of others.
Here's a chat room.

We can talk about the tasks we want to work on here.
And set a development path for ourselves.

> This conference system is not yet ready, we need the help of others.  **Join team by send and introduce yourself in a message at [this issue](https://github.com/On-System/On/issues/2)**

Study and research can be one of the things that will always be needed in this project.

Research fields related to this Project:
- WebRtc
- Websocket
- UDP/TCP Ports
- deep knowledge in network

Please state if something goes wrong.
I appreciate you.

**Comment by BaseMax** [comment]
What's STUN? What's TURN?

![photo_2020-12-07_21-19-56](https://user-images.githubusercontent.com/2658040/101386152-00a47e80-38d2-11eb-9081-327b9a9e3779.jpg)

![photo_2020-12-07_21-19-56](https://user-images.githubusercontent.com/2658040/101386180-0ac67d00-38d2-11eb-9e6a-aca629139a34.jpg)

**Comment by BaseMax** [comment]
**PeerJS:**
Simple peer-to-peer with WebRTC

https://peerjs.com/

**Reply by HamzaAnis** [reply]
WebRTC would be good for the video conference?

**Reply by BaseMax** [reply]
Yes, WebRTC is good. @HamzaAnis 
We will want to develop a html5-base meeting platform. (Not flash)

**Comment by BaseMax** [comment]
List of some free STUN server:

- https://gist.github.com/mondain/b0ec1cf5f60ae726202e
- https://gist.github.com/zziuni/3741933
- etc (You can share more if you know a great STUN server)

**Comment by BaseMax** [comment]
How use STUN in peerjs?

example:

```javascript
var peer = new Peer({
	host: '192.168.1.14',
	port: 8878,
	path: '/peerjs',
	config: { 'iceServers': [
		{ url: 'stun:stun01.sipphone.com' },
		{ url: 'stun:stun.ekiga.net' },
		{ url: 'stun:stun.fwdnet.net' },
		{ url: 'stun:stun.ideasip.com' },
		{ url: 'stun:stun.iptel.org' },
		{ url: 'stun:stun.rixtelecom.se' },
		{ url: 'stun:stun.schlund.de' },
		{ url: 'stun:stun.l.google.com:19302' },
		{ url: 'stun:stun1.l.google.com:19302' },
		{ url: 'stun:stun2.l.google.com:19302' },
		{ url: 'stun:stun3.l.google.com:19302' },
		{ url: 'stun:stun4.l.google.com:19302' },
		{ url: 'stun:stunserver.org' },
		{ url: 'stun:stun.softjoys.com' },
		{ url: 'stun:stun.voiparound.com' },
		{ url: 'stun:stun.voipbuster.com' },
		{ url: 'stun:stun.voipstunt.com' },
		{ url: 'stun:stun.voxgratia.org' },
		{ url: 'stun:stun.xten.com' },
		{
			url: 'turn:192.158.29.39:3478?transport=udp',
			credential: 'JZEOEt2V3Qb0y27GRntt2u2PAYA=',
			username: '28224511:1379330808'
		},
		{
			url: 'turn:192.158.29.39:3478?transport=tcp',
			credential: 'JZEOEt2V3Qb0y27GRntt2u2PAYA=',
			username: '28224511:1379330808'
		}
	]
},
```

**Comment by BaseMax** [comment]
**EasyRTC:**
The fastest way to build your own WebRTC apps

https://easyrtc.com/

**Comment by BaseMax** [comment]
## PeerJs vs EasyRTC vs SimpleWebRTC

Which are trend?

https://www.npmtrends.com/easyrtc-vs-peerjs-vs-simple-peer-vs-simplewebrtc

**Comment by BaseMax** [comment]
A preview UI demo:

![image_2020-11-20_16-32-00](https://user-images.githubusercontent.com/2658040/101387263-72c99300-38d3-11eb-9e49-5a737d93284b.png)

**Comment by BaseMax** [comment]
Show new messages similar to Instagram:

![photo_2020-12-07_21-30-43](https://user-images.githubusercontent.com/2658040/101387306-8248dc00-38d3-11eb-9720-cf85536f9487.jpg)


timer: `fadeOut` animation and hidden/remove element...

**Comment by BaseMax** [comment]
My favorite UI for room/meeting:
![image_2020-11-20_12-38-45](https://user-images.githubusercontent.com/2658040/101387697-1024c700-38d4-11eb-885f-f8d7b4f467e9.png)

**Comment by BaseMax** [comment]
**Similar Tools:**

- Adobe Connect: https://www.adobe.com/products/adobeconnect.html
- BigBlueButton: https://bigbluebutton.org/
- Jitsi: https://meet.jit.si/
- Google Meet: https://meet.google.com/
- Google Duo: https://duo.google.com/about/
- Skype: https://web.skype.com/
- Microsoft Teams: https://www.microsoft.com/en-us/microsoft-365/microsoft-teams/group-chat-software

**Comment by HamzaAnis** [comment]
Any new update on it?

**Comment by BaseMax** [comment]
Hello Hamza,
Thank you for your message. @HamzaAnis

Unfortunately, one of the members became ill. I hope they recover as soon as possible.


I have sent my preferred UI.
Please write here if you have an comment.
Otherwise, does anyone want to develop this page? (HTML, CSS, JS)
https://github.com/On-System/On/issues/1#issuecomment-740084701

**Comment by BaseMax** [comment]
Do you agree and want to start develop room page? @HamzaAnis 

If yes, What framework and structure do you suggest?
Pure JS or ReactJs or Vuejs, etc?

What frameworks do you experience?

Thanks in advance.

**Comment by HamzaAnis** [comment]
I am experienced in React, Golang, Node. You can choose the stack about what should be best for it.

**Comment by HamzaAnis** [comment]
> My favorite UI for room/meeting:
> ![image_2020-11-20_12-38-45](https://user-images.githubusercontent.com/2658040/101387697-1024c700-38d4-11eb-885f-f8d7b4f467e9.png)

That look great.

**Reply by dhruval-tech** [reply]
@BaseMax Hey, I've created this UI some week ago, Not completed but if we want to go with this UI then We can do it.

I've attached a screenshot.
![Screenshot (692)](https://user-images.githubusercontent.com/55456254/102233561-6cac6580-3f16-11eb-9cac-35f13c5ab7ce.png)

I've put <iframe> tag just for temporary to check how it looks, We can change it.

**Reply by BaseMax** [reply]
Thanks, You can commit this or send Pull request. Next you modify style to make this similar to selected UI. @dhruval-tech

**Reply by dhruval-tech** [reply]
@BaseMax Sure.

**Comment by HamzaAnis** [comment]
I'll try to work on it.

**Comment by BaseMax** [comment]
Well, it would be great if you start develop the page with HTML, CSS and JS + React Js.
Other friends will join you later.

**Comment by HamzaAnis** [comment]
Do you have any gitter or slack channel? Communication over there would be a lot better.

**Comment by BaseMax** [comment]
GitHub works great for me here.
I will receive your messages immediately.
If others want, I agree to form a group there.

**Comment by BaseMax** [comment]
**Edit:** If you add yourself as watcher, you will get all message in email inbox.

If you know about the GitHub Discussion section. We may move there soon. It can be somewhat helpful.

Anyway, I agree with you.

**Comment by HamzaAnis** [comment]
OK sounds good. Yeah GitHub discussion would be better.

**Comment by BaseMax** [comment]
Okay, are you starting now?

You can clone or fork this,
Please send a Pull Request to this repository every step of the way. (Pull Request)

e.g first commit: create reactjs project

Thanks

**Comment by HamzaAnis** [comment]
Yes I understand the requirements. I'll work on it and get someone to work on it on our spare time. Is that fine? 
Also do create a Project Board for it. It would be helpful.

**Comment by HamzaAnis** [comment]
I am not the maintainer so I can not create it.

**Comment by BaseMax** [comment]
> I am not the maintainer so I can not create it.

You are a member in this org.

**Comment by BaseMax** [comment]
> Also do create a Project Board for it. It would be helpful.

Sure, thanks. I create a board for this.

Please check out this. Do you access to board?
https://github.com/On-System/On/projects/1?fullscreen=true

**Comment by HamzaAnis** [comment]
Yes I am but I don't have invite to this repo. Somehow I can not create or update board, maybe I don't have access for it.

**Comment by BaseMax** [comment]
What now?

**Comment by BaseMax** [comment]
> I'll work on it and get someone to work on it on our spare time. Is that fine?

Do you have a specific person in mind?

It will be great if you implement the page of the room / meeting.
We can comment on it with other friends and improve it.

We are now focusing only on the page and user interface.
We will work on the socket and network later.

**Comment by HamzaAnis** [comment]
> What now?

Yes it is good now.

**Comment by BaseMax** [comment]
Hi, What's up? @HamzaAnis

**Comment by BaseMax** [comment]
![](https://cdn.dribbble.com/users/4248396/screenshots/11158116/media/069811eed565f091b2335bea5182eceb.png)

**Comment by BaseMax** [comment]
![](https://cdn.dribbble.com/users/380990/screenshots/14141715/media/f0575917eb372c42b8f8cdce2d334bd9.png)

**Comment by BaseMax** [comment]
![](https://cdn.dribbble.com/users/455670/screenshots/13976106/media/545eb5589ef8c73ba6d7380519e9a3e7.png)

**Comment by BaseMax** [comment]
![](https://cdn.dribbble.com/users/1369035/screenshots/11702052/media/722e1996d383b41c1a567c89962fb74e.png)

**Comment by BaseMax** [comment]
![](https://cdn.dribbble.com/users/380990/screenshots/14143932/media/081a215971647f24d69c3e773ffa06ef.png)

**Comment by BaseMax** [comment]
![](https://cdn.dribbble.com/users/3381045/screenshots/13872896/media/114d74856edd7d0537c03fe43f411098.png)
![](https://cdn.dribbble.com/users/3381045/screenshots/13872896/media/a2d79d36b1418af40ca0fdaf8ffe5902.png)

**Comment by BaseMax** [comment]
![](https://cdn.dribbble.com/users/1198718/screenshots/14649929/media/b7de47e9d4af76c1f84f54385779029c.png)

**Comment by BaseMax** [comment]
![](https://cdn.dribbble.com/users/4654578/screenshots/11569150/media/5f49a7078f6a0fd2cbe3c23aef88e2a9.png)

**Comment by BaseMax** [comment]
![](https://cdn.dribbble.com/users/504106/screenshots/14243248/media/e13a2339fbc612de6f69f554e57dfd9c.png)

**Comment by BaseMax** [comment]
![](https://cdn.dribbble.com/users/504106/screenshots/13936650/media/f04b5b46ec91c9e3ec522ea493497f3a.png)

**Comment by BaseMax** [comment]
Some video:
- https://cdn.dribbble.com/users/148887/screenshots/13979070/media/b3f9cfdfb70b77627edbb230f6e26791.mp4
- https://cdn.dribbble.com/users/148887/screenshots/13979090/media/34169a8ca37710daa28902ce6a9c15ec.mp4
- https://cdn.dribbble.com/users/148887/screenshots/13979075/media/eb13cb09a4c1be5a2ddf3744d31d6db0.mp4

**Comment by BaseMax** [comment]
This discussion is being migrated.
Now it's discussion, not issue. :+1:

**Comment by BaseMax** [comment]
## What is FreeSWITCH?

FreeSWITCH is a Software Defined Telecom Stack enabling the digital transformation from proprietary telecom switches to a versatile software implementation that runs on any commodity hardware. From a Raspberry PI to a multi-core server, FreeSWITCH can unlock the telecommunications potential of any device. Combined with our hosted cloud platform, SignalWire, FreeSWITCH can interconnect with the outside world and scale to any size.


https://freeswitch.com/

**BigBlueButton** also use **freeswitch** for transfer audio, etc.

**Comment by BaseMax** [comment]
First version of meeting page will be ready today? @HamzaAnis

**Reply by HamzaAnis** [reply]
https://github.com/On-System/On/discussions/1#discussioncomment-193688 Yes I will try tomorrow.

**Reply by BaseMax** [reply]
What up? @HamzaAnis

**Reply by BaseMax** [reply]
I was waiting for your response. @HamzaAnis

**Reply by muhammadumerfarooq** [reply]
> I was waiting for your response. @HamzaAnis

He's busy right now, for time being i'm working on it.

**Reply by BaseMax** [reply]
Hi there, what's up? @HamzaAnis @muhammadumerfarooq @dhruval-tech

**Reply by BaseMax** [reply]
> [#1 (comment)](https://github.com/On-System/On/discussions/1#discussioncomment-193688) Yes I will try tomorrow.
Hi, welcome back. @HamzaAnis

**Comment by BaseMax** [comment]
Hi guys,
Welcome @nafasebra
We hired a new person and he improved and redesigned the UI.

I will be publish new selected UI as soon...

**Comment by BaseMax** [comment]
Incomplete images for when the design was in progress:

![demo1](https://user-images.githubusercontent.com/2658040/109949798-0eab4500-7cf1-11eb-9e46-2c1cd77b4cc5.jpg)
![demo2](https://user-images.githubusercontent.com/2658040/109949802-10750880-7cf1-11eb-92f4-26f7795e3a27.jpg)
![demo3](https://user-images.githubusercontent.com/2658040/109949808-11a63580-7cf1-11eb-8e83-8c6e359c391f.jpg)

— Using by Adobe XD

**Comment by BaseMax** [comment]
Final UI image:

![demo4](https://user-images.githubusercontent.com/2658040/109949900-2f739a80-7cf1-11eb-9556-11c630b7fe64.jpg)
![demo5](https://user-images.githubusercontent.com/2658040/109949912-31d5f480-7cf1-11eb-9a87-a6b51d02a0a9.jpg)

— Using by Adobe XD

**Comment by BaseMax** [comment]
It is really unfortunate how slow our progress was; and not helping to project. I hope interact more contributes as soon...

Anyway, the design is complete.
A small part of the HTML,CSS,Js layout is also done.

It can be seen preview here:
https://on-system.github.io/On/client/room.html

and watch source code at:
https://github.com/On-System/On/tree/main/client

**Comment by BaseMax** [comment]
Many improve have been made to page for supporting different devices size.
**However, there are still many small problems.**

You can see the latest version here:
https://on-system.github.io/On/client/room.html
